### DISCUSSION TITLE
I can't send more than 255 bytes of data

### DISCUSSION THREAD

**Discussion by yacoubhanna** [eProsima/Fast-DDS#3877] [Unofficial support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

I looked online about the maximum size of the message and it should be **up to 46k bytes**. However, I was not able to send that much data using the HelloWorld Example and running them on the same machine. You can see that the publisher sending all 10 messages but not on the subscriber side, they are not received, or at least they are not showing up on the terminal. Also, I ran Wireshark to capture the data that I was sending and I was not able to see it. I can only see the packet when someone subscribes to a topic.  

### Current behavior

Using a HelloWorld Example I was only able to send **255 bytes as a message**. I'm running the Publisher and the Subscriber on the same machine (Loopback) on Ubuntu 22.04.

### Steps to reproduce

.....

### Fast DDS version/commit

I'm using the current version 2.7.1

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

Default configuration, UDPv4 & SHM

### Additional context

Ubuntu Jammy 22.04 amd64

**Comment by methylDragon** [comment]
Are you setting a string? If you're setting a string, the default bound for the string length is 255.

[create_string_builder](https://www.eprosima.com/docs/fast-rtps/2.3.0/html/classeprosima_1_1fastrtps_1_1types_1_1_dynamic_type_builder_factory.html#a8a00a390ec249dd1dbbd04415d6fb68d) (uint32_t bound=[MAX_STRING_LENGTH](https://www.eprosima.com/docs/fast-rtps/2.3.0/html/namespaceeprosima_1_1fastrtps_1_1types.html#af7a3e2e0700a3624cb600b77600efe74))

[const int32_t MAX_STRING_LENGTH = 255](https://www.eprosima.com/docs/fast-rtps/2.3.0/html/namespaceeprosima_1_1fastrtps_1_1types.html#af7a3e2e0700a3624cb600b77600efe74)

So just set a different bound.

**Comment by yacoubhanna** [comment]
Thank you so much @methylDragon. I just have another question. I was able to run the secure and the insecure HelloworldExample. I ran Wireshark in order to capture the data, but unfortunately, I was not able to see the message that I was sending to the subscriber. Even though, the subscriber was receiving the message.

**Comment by EduPonz** [comment]
Hi @yacoubhanna,

That is because Fast DDS uses a SHM transport by default when it can. You could enforce the use of a UDP transport as explained [here](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/udp/udp.html#enabling-udp-transport). You also have the possibility of making the SHM transport to create a capture file [by setting dump file](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/shared_memory/shared_memory.html#sharedmemtransportdescriptor) that you can then inspect with wireshark.

**Comment by yacoubhanna** [comment]
Thank you @EduPonz, I was able to capture the data by having dump file and inspect it. However, I was wondering since I'm running the SecureHelloWorldExample, and using the "Crypto:AES-GCM-GMAC" Plugin. How can I access the keys that are used in that plugin. Also, how can I see the cipher message when I run the publisher and the subscriber terminals? When I run the publisher I can see the message that I sent, but I need to see how the message looks encrypted. 

Thanks for your help @methylDragon @EduPonz

**Comment by yangrudan** [comment]
I used shm transport and send string more than 255bytes, using the way below (picture is C++ and python version):
![a780b574a942067720d733e4d1feedad](https://github.com/eProsima/Fast-DDS/assets/66586788/e66a5d60-f386-4751-a39a-bbe4ff978366)
![62b0b27dbd58a8714125b228ff731aa1](https://github.com/eProsima/Fast-DDS/assets/66586788/58cd0ace-5394-4fa8-a9c5-26f8f0c3c8d8)
![6ff752f05dee702079fde69513ebcfe5](https://github.com/eProsima/Fast-DDS/assets/66586788/a108c9f8-3cda-4242-83bd-d1bf9a5cae8b)
Hope useful.

**Comment by JesusPoderoso** [comment]
Hi @yacoubhanna, sorry for the late response.

> I was wondering since I'm running the SecureHelloWorldExample, and using the "Crypto:AES-GCM-GMAC" Plugin. How can I access the keys that are used in that plugin. 

Unfortunately, there is no way to access the keys.

> Also, how can I see the cipher message when I run the publisher and the subscriber terminals? When I run the publisher I can see the message that I sent, but I need to see how the message looks encrypted.

As done in Wireshark, you can use [tshark](https://www.wireshark.org/docs/man-pages/tshark.html) to analyze network traffic in terminal instead of graphical interface. Hope it suits your needs.

I am moving the issue to the Unofficial support discussion forum for the moment as per [Fast DDS Contributing](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md)
